presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -259,6 +261,266 @@ Field           Required  Type      Description
 
 There is no limit on field descriptions for either key or message.
 
+Kafka inserts
+-------------
+
+The Kafka connector supports the use of SQL ``INSERT INTO`` statements to write data to a Kafka topic.
+Table column data is mapped to Kafka messages as defined in the table definition file, `Table Definition Files`_.
+There are currently three supported data formats for key and message encoding, raw, CSV, and Avro. These data
+formats each have an encoder that handles mapping table column values.
+
+Example ``INSERT INTO`` query for a hypothetical "product" table with fields name (VARCHAR), and key (BIGINT):
+
+.. code-block:: SQL
+
+    INSERT INTO product (name, key) VALUES ('product name', 12345);
+
+Row encoding
+------------
+
+Encoding allows SQL insert statements to be used and defines how table columns in Presto
+map to Kafka key and message data.
+
+The Kafka connector contains the following encoders:
+
+* raw encoder - Table columns are mapped to Kafka message as raw bytes
+* CSV encoder - Kafka message is formatted as comma separated message, and fields are mapped to table columns
+* Avro encoder - Kafka message is mapped to Avro fields based on an Avro schema
+
+.. note::
+
+    A table definition file must be defined for the encoder to work. `Table Definition Files`_
+
+raw encoder
+^^^^^^^^^^^^^^^
+
+The raw encoder formats the table columns as raw bytes using the mapping information specified in the
+table definition file, `Table Definition Files`_.
+
+For fields, the following attributes are supported:
+
+* ``dataFormat`` - Selects the width of the column data type
+* ``type`` - Presto data type
+* ``mapping`` - ``<start>[:<end>]``; start and end position of bytes to convert
+
+The ``dataFormat`` attribute selects the number of bytes converted.
+If absent, ``BYTE`` is assumed. All values are signed.
+
+Supported values are:
+
+* ``BYTE`` - one byte
+* ``SHORT`` - two bytes (big-endian)
+* ``INT`` - four bytes (big-endian)
+* ``LONG`` - eight bytes (big-endian)
+* ``FLOAT`` - four bytes (IEEE 754 format)
+* ``DOUBLE`` - eight bytes (IEEE 754 format)
+
+The ``type`` attribute defines the Presto data type.
+
+Depending on Presto type assigned to column different values of dataFormat can be used:
```

Comment:
Depending on the Presto type assigned to a column different values of ``dataFormat`` can be used: